File: scripts/list_buckets.js
const { Storage } = require('@google-cloud/storage');
const fs = require('fs');
const path = require('path');

const serviceAccountPath = path.join(__dirname, '../green-force-pwa-2025-firebase-adminsdk-fbsvc-01d92f9a8e.json');

const storage = new Storage({ keyFilename: serviceAccountPath });

async function listBuckets() {
    try {
        const [buckets] = await storage.getBuckets();
        console.log('Buckets en este proyecto GCP:');
        buckets.forEach(bucket => {
            console.log(bucket.name);
        });
    } catch (error) {
        console.error('Error:', error.message);
    }
}

listBuckets();

listBuckets();

File: scripts/list_buckets.py
import json
import requests
from google.oauth2 import service_account
from google.auth.transport.requests import Request

key_file = "green-force-pwa-2025-firebase-adminsdk-fbsvc-01d92f9a8e.json"
creds = service_account.Credentials.from_service_account_file(
    key_file,
    scopes=["https://www.googleapis.com/auth/cloud-platform"]
)

creds.refresh(Request())
token = creds.token

project_id = "green-force-pwa-2025"
url = f"https://storage.googleapis.com/storage/v1/b?project={project_id}"

headers = {"Authorization": f"Bearer {token}"}
r = requests.get(url, headers=headers)

print("BUCKETS LIST STATUS:", r.status_code)
print("RESPONSE:", json.dumps(r.json(), indent=2))
